src/pkg/packager/filters/select_test.go

```diff
@@ -0,0 +1,91 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: 2021-Present The Zarf Authors
+
+// Package filters contains core implementations of the ComponentFilterStrategy interface.
+package filters
+
+import (
+	"testing"
+
+	"github.com/defenseunicorns/zarf/src/types"
+	"github.com/stretchr/testify/require"
+)
+
+func Test_selectStateFilter_Apply(t *testing.T) {
+	tests := []struct {
+		name                string
+		requestedComponents string
+		components          []types.ZarfComponent
+		expectedResult      []types.ZarfComponent
+		expectedError       error
+	}{
+		{
+			name:                "Test when requestedComponents is empty",
+			requestedComponents: "",
+			components: []types.ZarfComponent{
+				{Name: "component1"},
+				{Name: "component2"},
+				{Name: "component3"},
+			},
+			expectedResult: []types.ZarfComponent{
+				{Name: "component1"},
+				{Name: "component2"},
+				{Name: "component3"},
+			},
+			expectedError: nil,
+		},
+		{
+			name:                "Test when requestedComponents contains a valid component name",
+			requestedComponents: "component2",
+			components: []types.ZarfComponent{
+				{Name: "component1"},
+				{Name: "component2"},
+				{Name: "component3"},
+			},
+			expectedResult: []types.ZarfComponent{
+				{Name: "component2"},
+			},
+			expectedError: nil,
+		},
+		{
+			name:                "Test when requestedComponents contains an excluded component name",
+			requestedComponents: "component1, -component2",
```

Comment:
This would be better if the inclusion included the exclusion:
```suggestion
			requestedComponents: "comp*, -component2",
```